write up of 2026 large data track (#201)
neuroinformatics-unit/neuroinformatics-unit.github.io: docs/source/open-software-summer-school/2026/large-data.md

(track-large-array-data-2026)=
# Track: Large Array Data

> In many areas of research, large array datasets have become important and wide-spread. Examples include whole-organ images (e.g. light-sheet, HiP-CT), Neuropixels data, and long behavioural videos — to name just a few. Processing, analyzing and visualising these datasets is made significantly more complex and slow because they do not fit into a typical computer's memory and take up a lot of disk space.

Reviewer: Just to make the abstraction level of each data type consistent. 
```suggestion
In many areas of research, large array datasets have become important and wide-spread. Examples include whole-organ images, extracellular electrophysiology, and long behavioural videos — to name just a few. Processing, analyzing and visualising these datasets is made significantly more complex and slow because they do not fit into a typical computer's memory and take up a lot of disk space.
```

> The use of compression algorithms and "chunked" file formats (which favour parallel processing) enable us to deal with these challenges effectively, but doing this well requires careful consideration of the trade-offs between read and write speeds, disk space used, and portability.

Reviewer 2: ```suggestion
The use of compression algorithms and modern file formats which favour parallel processing enable us to deal with these challenges effectively, but doing this well requires careful consideration of the trade-offs between read and write speeds, disk space used, and portability.
```
A suggestion to avoid "chunked"

> This track will cover the necessary context needed to understand the computational challenges posed by large array datasets, and how to address these challenges using open-source software tools. All sessions will include a significant hands-on component.

Reviewer: Could we make it a bit clearer what participants will be able to by the end of the week? "Process a 1TB functional imaging dataset on their laptop" or things like that?

::: {admonition} Target audience
:class: note

> This course is designed for researchers and students interested in learning about open-source tools for processing large array data with Python. Participants are encouraged to bring their own data to experiment with, but example data will be provided.

Reviewer: Should we make it clearer that it is for people who have, or expect to acquire large datasets? Currently "an immediate use case" is one of our selection criteria.

:::

## Course overview

### Core workshop (Monday - Wednesday)

We will cover the following topics during the first three days:

> * **Motivation** We will show examples of large array datasets from neuroscience and imaging, discuss what research they enable, in what situations you should acquire them, and the computational challenges they pose.

Author: We plan to extend to examples from electrophysiology which are not images but also arrays :)

* **Skill building:** we'll start with a practical workshop on **Git and GitHub** to equip everyone with the necessary skills for collaborative coding.
* **Project-based work:** participants will self-organize into small teams to tackle projects hands-on. **Coding is not a requirement**; any idea that benefits from collaboration with other attendees is welcome. Potential project ideas include, but are not limited to:
    * **Apply a tool:** use any learned software to analyze a new dataset (your own or a public one).
    * **Give feedback:** raise issues, suggest features, or improve documentation for relevant open-source tools.
    * **Make a contribution:** submit a pull request to an open-source repository (support will be provided).
    * **Collaborative writing:** draft a white paper, blog post, or documentation together.
    * **Prototype an idea:** test a cool new analysis or method on real-world data.
* **Presentation:** you will have the opportunity to report your team's progress on the final afternoon.
* 
## Instructors
- [Alessandro Felder](https://github.com/alessandrofelder)
- [Igor Tatarnikov](https://github.com/IgorTatarnikov)

> - [David Stansby](https://github.com/dstanby)

Reviewer: ```suggestion
- [David Stansby](https://github.com/dstansby)
```
